core/src/main/java/com/google/zxing/datamatrix/encoder/HighLevelEncoder.java

```diff
@@ -295,32 +295,34 @@ static int lookAheadTest(CharSequence msg, int startpos, int currentMode) {
         int[] intCharCounts = new int[6];
         byte[] mins = new byte[6];
         findMinimums(charCounts, intCharCounts, Integer.MAX_VALUE, mins);
-        int minCount = getMinimumCount(mins);
 
-        if (intCharCounts[ASCII_ENCODATION] < intCharCounts[BASE256_ENCODATION]
-            && intCharCounts[ASCII_ENCODATION] < intCharCounts[C40_ENCODATION]
-            && intCharCounts[ASCII_ENCODATION] < intCharCounts[TEXT_ENCODATION]
-            && intCharCounts[ASCII_ENCODATION] < intCharCounts[X12_ENCODATION]
-            && intCharCounts[ASCII_ENCODATION] < intCharCounts[EDIFACT_ENCODATION]) {
+        if (intCharCounts[ASCII_ENCODATION] < min(intCharCounts[BASE256_ENCODATION]
```

Comment:
So there's no "+ 1" with "<" in this case, unlike the others?
It feels like we could refactor this further - given an index in intCharCounts, say whether it's more than 1 smaller than other indices?

Reply:
Well in the spec all is written as a + 1 < b or a + 1 <= b. But as you pointed out, a + 1 <= b can be rewritten as a < b so I changed that. 
The situation is as follows:
We have one case where we compare +1 <= against all others.
Then we have a second case where we compare <= against one or +1 < against 4 other values
Then we have three cases where we compare + 1 < against 5 other values and finally a case where we compare + 1 < against 4 other values.
Since sometimes you have to compare against 4 and sometimes against 5 values, I don't see how the signature of a function that works with indexes into intCharCounts could look good and understandable. Maybe I am overlooking something. Can you be more concrete?

Reply:
OK I get it, so "+ 1 <" is intended, just checking.
OK if the logic varies that much, this may be as good as it gets. The min() function looks a little clunky and could be written as varargs instead... but may be a little slower, so, no big deal